spyder_kernels/utils/iofuncs.py

```diff
@@ -375,7 +375,26 @@ def load_dictionary(filename):
     error_message = None
     try:
         with tarfile.open(filename, "r") as tar:
-            tar.extractall()
+            def is_within_directory(directory, target):
+                
+                abs_directory = os.path.abspath(directory)
+                abs_target = os.path.abspath(target)
+            
+                prefix = os.path.commonprefix([abs_directory, abs_target])
+                
+                return prefix == abs_directory
+            
+            def safe_extract(tar, path=".", members=None, *, numeric_owner=False):
+            
+                for member in tar.getmembers():
+                    member_path = os.path.join(path, member.name)
+                    if not is_within_directory(path, member_path):
+                        raise Exception("Attempted Path Traversal in Tar File")
```

Comment:
```suggestion
                        raise Exception(f"Attempted Path Traversal in Tar File {tar.name!r}")
```

Seems like it would be worth mentioning the name of the suspect file in the error message